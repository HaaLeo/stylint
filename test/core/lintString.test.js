'use strict';

const stylint = require('../../index');

const app = stylint().create();

// turn on strict mode from this point and turn off unnecessary logging
app.state.quiet = true;
app.state.watching = true;

describe('Lint Text: ', () => {
  let linter;

  beforeEach(() => {
    linter = stylint.api();
  });

  it('should return object with violations', () => {
    expect(linter.lintString('.class {\n  color: red !important\n}\n', null, 'filename.styl')).toMatchSnapshot();
  });
});
