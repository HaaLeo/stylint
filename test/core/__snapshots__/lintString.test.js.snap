// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lint Text:  should return object with violations 1`] = `
Object {
  "errorCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "filename.styl",
      "messages": Array [
        Object {
          "column": 13,
          "line": 2,
          "message": "!important is disallowed",
          "ruleId": "noImportant",
          "severity": "warning",
          "source": "  color: red !important",
        },
        Object {
          "column": 23,
          "line": 2,
          "message": "missing semicolon",
          "ruleId": "semicolons",
          "severity": "warning",
          "source": "  color: red !important",
        },
      ],
      "warningCount": 2,
    },
  ],
  "warningCount": 2,
}
`;
